addons/website/static/src/snippets/s_rating/options.js

```diff
@@ -0,0 +1,160 @@
+odoo.define('website.s_rating_options', function (require) {
+'use strict';
+
+const weWidgets = require('wysiwyg.widgets');
+const options = require('web_editor.snippets.options');
+
+options.registry.Rating = options.Class.extend({
+    /**
+     * @override
+     */
+    start: function () {
+        this.faClassActiveCustomIcons = '';
+        this.faClassInactiveCustomIcons = '';
+        this.iconType = this.$target[0].dataset.icon;
+        this.nbActiveIcons = this.$target.find('.s_rating_active_icons > i').length;
+        this.nbTotalIcons = this.$target.find('.s_rating_icons i').length;
+        if (this.iconType === 'custom') {
+            this.faClassActiveCustomIcons = this.$target[0].dataset.activeCustomIcon;
+            this.faClassInactiveCustomIcons = this.$target[0].dataset.inactiveCustomIcon;
+        }
+        return this._super.apply(this, arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Display the selected icon type.
+     *
+     * @see this.selectClass for parameters
+     */
+    setIcons: function (previewMode, widgetValue, params) {
+        this.iconType = widgetValue;
+        this._renderIcons();
+        this.$target[0].dataset.icon = widgetValue;
+        this.$target[0].dataset.activeCustomIcon = '';
+        this.$target[0].dataset.inactiveCustomIcon = '';
+    },
+    /**
+     * Allows to select a font awesome icon with media dialog.
+     *
+     * @see this.selectClass for parameters
+     */
+    customIcon: async function (previewMode, widgetValue, params) {
+        return new Promise(resolve => {
+            const dialog = new weWidgets.MediaDialog(
+                this,
+                {noImages: true, noDocuments: true, noVideos: true, mediaWidth: 1920},
+                $('<i/>')
+            );
+            this._saving = false;
+            dialog.on('save', this, function (attachments) {
+                this._saving = true;
+                const customClass = 'fa fa-fw ' + attachments.className;
+                let $activeIcons = this.$target.find('.s_rating_active_icons > i');
+                let $inactiveIcons = this.$target.find('.s_rating_inactive_icons > i');
+                if (params.customActiveIcon === 'true') {
+                    $activeIcons.removeClass().addClass(customClass);
+                } else {
+                    $inactiveIcons.removeClass().addClass(customClass);
+                }
```

Comment:
can simplify that:
```javascript
const $icons = params.customActiveIcon ? $active : $inactive;
$icon.removeClass().addClass(customClass);
```

Also use `const` instead of `let`